src/test/java/spoon/reflect/visitor/CtInheritanceScannerTest.java

```diff
@@ -65,23 +73,29 @@
 			if (!intf.getSimpleName().startsWith("Ct")) {
 				continue;
 			}
-			Method mth;
-			try {
+			Method mth=null;
+			
+			// if a method visitX exists, it must be invoked
+			try {				
 				mth = CtInheritanceScanner.class.getDeclaredMethod("visit" + intf.getSimpleName(), intf);
-
-				// we should not have a scan method
-				try {
-					CtInheritanceScanner.class.getDeclaredMethod("scan" + intf.getSimpleName(), intf);
-					Assert.fail("should not have both visit" + intf.getSimpleName() + " and scan" + intf.getSimpleName()
```

Comment:
You didn't check no more if we have both visit and scan methods for one interface.